portafly/src/components/util/AlertsContext.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { useContext } from 'react'
+import {
+  AlertGroup,
+  AlertActionCloseButton,
+  Alert,
+  AlertVariant
+} from '@patternfly/react-core'
+
+interface IAlert {
+  id: string
+  variant: keyof typeof AlertVariant
+  title: string
+  timeout: NodeJS.Timeout
+}
+
+type IAlertsContext = { addAlert: (alert: Omit<IAlert, 'timeout'>) => void }
+const AlertsContext = React.createContext<IAlertsContext>({ addAlert: () => {} })
+
+const AlertsProvider: React.FunctionComponent = ({ children }) => {
+  const [alerts, setAlerts] = React.useState<IAlert[]>([])
+
+  const removeAlert = (id: string) => {
+    setAlerts((prevAlerts) => prevAlerts.filter((a) => a.id !== id))
+  }
+
+  const addAlert = (alert: Omit<IAlert, 'timeout'>) => {
+    const timeout = setTimeout(() => removeAlert(alert.id), 8000)
+    setAlerts((prevAlerts) => [...prevAlerts, { ...alert, timeout }])
+  }
```

Comment:
Not super important, but for the sake of composition, something like...

```suggestion
const TIMEOUT = 8000
...
const alertTimeout = (id: string) => setTimeout(() => removeAlert(id), TIMEOUT)

const addAlert (alert: Omit<IAlert, 'timeout'>) => (
  setAlerts((prevAlerts) => [...prevAlerts, { ...alert, alertTimeout(alert.id) }])
)
```

Reply:
I extracted `TIMEOUT` as a constant but I don't see the value of having a function for `alertTimeout`. It belongs only to `addAlert` IMO.